CoffeeShop/src/main/java/order/OrderList.java

```diff
@@ -0,0 +1,82 @@
+package order;
+
+import interfaces.EntityList;
+
+import java.util.ArrayDeque;
+import java.util.Queue;
+import java.util.UUID;
+
+/**
+ * @author Fraser Holman
+ *
+ * Class represents a list of current orders
+ *
+ * Contains a queue of different orders created by customers
+ */
+
+public class OrderList implements EntityList<Order, UUID> {
+    private Queue<Order> queue;
+
+    /**
+     * Initialises the queue to contain all the orders
+     */
+    public OrderList() {
+        queue = new ArrayDeque<Order>();
+    }
+
+    /**
+     * Adds an order to the queue of orders
+     *
+     * @param order The order to be added to the queue
+     */
+    @Override
+    public void add(Order order) {
+        queue.add(order);
+    }
+
+    /**
+     * Removes an order from the queue of orders for processing
+     *
+     * @param ID The ID used to find the order to be removed
+     */
+    @Override
+    public void remove(UUID ID) {
+        for (Order o : queue) {
+            if (o.getOrderID().equals(ID)) {
+                queue.remove(o);
+                return;
+            }
+        }
+        throw new IllegalArgumentException(ID + " is not a valid order ID");
+    }
```

Comment:
```suggestion
    @Override
    public boolean remove(UUID ID) {
        return queue.removeIf(order -> order.getOrderID().equals(ID));
    }
```

we can simplify like this, this will return true if the item is successfully removed or false if not, then wherever we call this is where we can handle the exception if it is not removed. This is called letting the exception bubble up

more info on removeIf: https://stackoverflow.com/questions/43246613/java-removeif-example